src/aquarium-optimized/Aquarium.cpp

```diff
@@ -786,42 +785,38 @@ void Aquarium::render() {
       resetFpsTime();
     }
 
-  bool updateAndDrawForEachFish =
-      toggleBitset.test(static_cast<size_t>(TOGGLE::UPATEANDDRAWFOREACHMODEL));
-
-  if (updateAndDrawForEachFish) {
-    updateAndDrawBackground();
-    updateAndDrawFishes();
-    mContext->updateFPS(mFpsTimer, &mCurFishCount, &toggleBitset);
-  } else {
-    updateBackground();
-    updateFishes();
-    mContext->updateFPS(mFpsTimer, &mCurFishCount, &toggleBitset);
-
-    // Begin render pass
+  updateAndDrawBackground();
+  updateAndDrawFishes();
```

Comment:
When `mDrawPerInstance` is `false` they don't really draw, but the names sound overlapping with the following `draw{Background,Fishes}()`. Let's consider a less confusing proposal for them.

Reply:
Not sure if the following is possible:

```c++
void Aquarium::updateFishes(UpdateCallback cb) {
  // ...
}

void Aquarium::render() {
  // ...
  if (mDrawPerInstance) {
    updateFishes(drawFishTrampoline);
  } else {
    updateFishes(nullptr);
    drawFishes();
  }
  // ...
}
```

Reply:
I will further refine this part in next patches.